### PR TITLE
gpcheckcat: Add new option -x to set session level GUCs

### PR DESCRIPTION
**Pull request by kpraveen457**
Problem : Running gpcheckcat FK check taking a lot of time to complete, One of the workarounds to fix this is to Pull the FK query and run it separately in psql using the GUC disable_cost='1e20' at session level. It help in making the query run faster but its not acceptable solution.So need to set the GUC at session level.

Solution : Added a new option [-x] in gpcheckcat that will provide support for PGOPTIONS to pass multiple session GUCs [such as disable_costs, gp_role] to set the parameter.

Implementation : Added a new option -x which is used to pass the multiple session GUCs to set the parameter. This is helpful when wanting to run certain tests in gpcheckcat with session level parameters. User can use the -x option multiple times to specify multiple GUC.

User can pass the GUC like : `gpcheckcat -x "disable_cost=3e15" -x "log_min_messages=debug5" -R foreign_key`
Note : GUCs with spaces are not allowed because PGOPTIONS currently don't support GUCs with spaces.

Added behave test case for the same

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Review by kainwen — APPROVED** [review]
LGTM. Please also have another approval from CM team. Thanks a lot!

**Comment by ashwinstar** [comment]
PR description example doesn't clarify hence asking explicitly, what are quoting requirements for this feature - means if GUC value has strings with spaces what way user has to specify?

**Comment by kpraveen457** [comment]
> PR description example doesn't clarify hence asking explicitly, what are quoting requirements for this feature - means if GUC value has strings with spaces what way user has to specify?

As PGOPTIONS currently don't support GUCs with spaces. Added this as a note in the PR description.

File: gpMgmt/bin/gpcheckcat [hunk 1]
```diff
@@ -328,6 +331,8 @@ def connect(user=None, password=None, host=None, port=None,
     options = '-c search_path='
     if utilityMode:
         options += ' -c gp_role=utility'
+    for x in GV.opt['-x']:
```
Comment (jnihal) [inline gpMgmt/bin/gpcheckcat:334]: nit: `x` -> `val`

**Review by jnihal — COMMENTED** [review]
Update the documentation section with the new option -https://github.com/greenplum-db/gpdb/blob/800bc609f36ccd53febe0e916f09b8fbc21d7882/gpMgmt/bin/gpcheckcat#L3-L20

File: gpMgmt/bin/gpcheckcat (hunk 1 above)
Comment (kpraveen457) [reply gpMgmt/bin/gpcheckcat]: done

**Review by jnihal — APPROVED** [review]

**Pull request merged**